sample_tpl/struct-padding.tpl

```diff
@@ -0,0 +1,52 @@
+struct State {
+    count  : uint32
+    sorter : Sorter
+    count1 : uint32
+    count2 : uint32
+}
+
+struct State2 {
+    count  : uint32
+    sorter : Sorter
+    count1 : [1]uint32
+    count2 : uint32
+}
+
+struct Embed {
+    a : int32
+}
+
+struct State3 {
+    count  : uint32
+    sorter : Sorter
+    embed  : Embed
+    count2 : uint32
+}
+
+fun test1() -> int32 {
+    var state: State
+    state.count2 = 500
+    state.count = 0
+    state.count1 = 42
+    return state.count2
+}
+
+fun test2() -> int32 {
+    var state: State2
+    state.count2 = 500
+    state.count1[0] = 42
+    return state.count2
+}
+
+fun test3() -> int32 {
+    var state: State3
+    state.count2 = 500
+    state.embed.a = 42
+    return state.count2
+}
+
+fun main() -> int32 {
+    var sum : int32
+    sum = test1() + test2() + test3()
+    return sum
+}
```

Comment:
Can you add some comments on what's going on here?

Reply:
Added comments.